File: easy/the_major_element.py
import sys
def process(line):
    L = line.count(",")
    d = {}
    for el in line.split(","):
        if el not in d:
            d[el] = 1
        else:
            d[el] += 1
            if d[el] >= L:
                print el
                return
    out = ""
    for e,v in d.iteritems():
        if v >= len(d)/2:
            print e
            return
    print "None"

with open(sys.argv[1],'r') as f:
    for line in f:
        process(line)

File: easy/the_major_element.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>

using namespace std;

void process(string line) {
    line += ",";
    int L = 0, i;
    for (i = 0; i < line.size(); i++) {
        if (line[i] == ',') {
            L++;
        }
    }
    string buf = "";
    map<string,int> data;
    for (i = 0; i < line.size(); i++) {
        if (line[i] == ',') {
            if (data.find(buf) == data.end()) {
                data[buf] = 1;
            } else {
                data[buf]++;
            }
            if (data[buf] >= L/2) {
                cout << buf << endl;
                return;
            }
            buf = "";
        } else {
            buf += line[i];
        }
    }
    cout << "None" << endl;
}

int main(int argc, char *argv[]) {
    ifstream f (argv[1]);
    string line;
    while (getline(f,line)) {
        process(line);
    }
    f.close();
    return 0;
}

